vault_contract/src/contract.rs

```diff
@@ -32,22 +35,33 @@ impl VaultTrait for VaultContract {
     }
 
     fn authorize_issuer(e: Env, admin: Address, issuer: Address, did: String) {
-        let contract_admin = storage::read_admin(&e);
-        if contract_admin != admin {
-            panic_with_error!(e, ContractError::NotAuthorized)
-        }
-        admin.require_auth();
+        validate_admin(&e, admin);
+        validate_did(&e, &did);
 
         issuer::authorize_issuer(&e, &issuer, &did);
     }
 
     fn revoke_issuer(e: Env, admin: Address, issuer: Address, did: String) {
-        let contract_admin = storage::read_admin(&e);
-        if contract_admin != admin {
-            panic_with_error!(e, ContractError::NotAuthorized)
-        }
-        admin.require_auth();
+        validate_admin(&e, admin);
+        validate_did(&e, &did);
 
         issuer::revoke_issuer(&e, &issuer, &did)
     }
 }
+
+fn validate_admin(e: &Env, admin: Address) {
+    let contract_admin = storage::read_admin(e);
+    if contract_admin != admin {
+        panic_with_error!(e, ContractError::NotAuthorized)
+    }
+    admin.require_auth();
+}
+
+fn validate_did(e: &Env, did: &String) {
+    if !did::is_registered(e, did) {
+        panic_with_error!(e, ContractError::DidNotFound)
+    }
+    if let Some(true) = did::is_revoked(e, did) {
+        panic_with_error!(e, ContractError::DidRevoked)
+    }
```

Comment:
It would be more optimal to just do one read from storage. Currently, there are 2, which will end up in more charged fee.